tests/rptest/clients/oauth_producer.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+# TODO: Different grant types
+# TODO: Metadata reads don't seem to work (same issue as AdminClient)
+
+import requests
+import time
+import functools
+
+from confluent_kafka import Producer
+
+class OAuthProducer:
```

Comment:
do we plan to integrate oauth support into other kafka clients we use in ducktape (at least the ones that support it)?

Reply:
Good question. There's a line item on the epic to add oauth support to a Java client, but that's it afaik.